pimd/pim_zlookup.c

```diff
@@ -254,32 +245,23 @@ static int zclient_read_nexthop(struct pim_instance *pim,
 			 * allow us to work in cases where we are
 			 * trying to find a route for this box.
 			 */
-			nexthop_tab[num_ifindex].nexthop_addr.family = AF_INET;
-			nexthop_tab[num_ifindex].nexthop_addr.prefixlen =
-				IPV4_MAX_BITLEN;
-			nexthop_tab[num_ifindex].nexthop_addr.u.prefix4 =
-				addr;
+			nexthop_tab[num_ifindex].nexthop_addr = addr;
 			++num_ifindex;
 			break;
 		case NEXTHOP_TYPE_IPV4_IFINDEX:
 		case NEXTHOP_TYPE_IPV4:
-			nexthop_tab[num_ifindex].nexthop_addr.family = AF_INET;
-			nexthop_tab[num_ifindex].nexthop_addr.u.prefix4.s_addr =
+			nexthop_tab[num_ifindex].nexthop_addr.s_addr =
 				stream_get_ipv4(s);
 			nexthop_tab[num_ifindex].ifindex = stream_getl(s);
 			++num_ifindex;
 			break;
 		case NEXTHOP_TYPE_IPV6_IFINDEX:
-			nexthop_tab[num_ifindex].nexthop_addr.family = AF_INET6;
-			stream_get(&nexthop_tab[num_ifindex]
-					    .nexthop_addr.u.prefix6,
-				   s, sizeof(struct in6_addr));
+			stream_get(&nexthop_tab[num_ifindex].nexthop_addr, s,
+				   sizeof(struct in6_addr));
 			nexthop_tab[num_ifindex].ifindex = stream_getl(s);
 
-			p.family = AF_INET6;
-			p.prefixlen = IPV6_MAX_BITLEN;
```

Comment:
Still need the 2 lines above here, otherwise `p` will have uninitialized fields.

Reply:
Sorry my bad. Fixed this